ci/run_ci.py

```diff
@@ -0,0 +1,107 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+import argparse
+import subprocess
+import platform
+import psutil
+import urllib.request as ulib
+import os
+
+
+def _exec_cmd(cmd: str):
+    print(f"run command: {cmd}")
+    result = subprocess.check_output(cmd, shell=True).decode("utf-8")
+    print(result)
+    return result
+
+
+def _get_os_name_lower():
+    return platform.system().lower()
+
+
+def _get_os_machine():
+    return platform.machine()
+
+
+def _install_bazel():
+    os_name = _get_os_name_lower()
+    if os_name == "windows":
+        # TODO: Implement Windows download bazel.
```

Comment:
I don't think it is hard to implement, you can see there are many executable for windows in this page:

https://github.com/bazelbuild/bazel/releases/6.3.2/

Reply:
Requires additional installation of Visual C++ Build Tool and setting environment variables.

https://bazel.build/configure/windows#build_cpp

Reply:
We can default to these things already being installed.

Reply:
Currently the download and installation logic is implemented here ?

Reply:
I mean we can assume MSVC are already installed.

Reply:
```python
def _install_bazel():
    is_windows = _get_os_name_lower() == "windows"
    download_url_base = f"https://github.com/bazelbuild/bazel/releases/download/{BAZEL_VERSION}"
    local_name = "bazel.exe" if is_windows else "bazel-installer.sh"
    exe_tail = "exe" if is_windows else "sh"
    bazel_download_url = f"{download_url_base}/bazel-{BAZEL_VERSION}{'' if is_windows else '-installer'}-" \
                         f"{os_name}-{_get_os_machine()}.{exe_tail}"
    ulib.urlretrieve(bazel_download_url, local_name)

    if is_windows:
        bazel_path = os.path.join(os.curdir, local_name)
        _exec_cmd(f"setx path \"%PATH%;{bazel_path}\"")
    else:
        os.chmod(local_name, 0o777)
        _exec_cmd(f"./{local_name} --user")
        os.remove(local_name)

    # bazel install status check
    _exec_cmd("bazel --version")

    # default is byte
    total_mem = psutil.virtual_memory().total
    limit_jobs = int(total_mem / 1024 / 1024 / 1024 / 3)
    with open(".bazelrc", "a") as file:
        file.write(f"\nbuild --jobs={limit_jobs}")


def _run_cpp():
    # run test
    query_result = _exec_cmd("bazel query //...")
    _exec_cmd("bazel test {}".format(query_result.replace("\n", " ").replace("\r", " ")))
```

Windows download has been implemented, PTAL.

Reply:
looks good to me.